make: grep for new go:build tags in PRIV_TEST_PKGS_EVAL

tklauser opened the pull request:
Updating to Go 1.18 and removing all +build tags in commit 1108684e354d
("Update Go to 1.18") lead to any make target reporting the following
error as reported by Quentin in [1]:

    $  make help > /dev/null
    dirname: missing operand
    Try 'dirname --help' for more information.

Fix this by instead grep'ing for the new go:build tags in
PRIV_TEST_PKGS_EVAL.

[1] https://github.com/cilium/cilium/pull/19169#issuecomment-1096858549

Fixes: 1108684e354d ("Update Go to 1.18")

qmonnet reviewed (approved):
> broke the `make help` target

I think it's for all targets, that was just an example I provided to check quickly and without too much output :).

Anyway. Looks good, thanks for the quick fix!

tklauser committed:
make: grep for new go:build tags in PRIV_TEST_PKGS_EVAL

Updating to Go 1.18 and removing all +build tags in commit 1108684e354d
("Update Go to 1.18") lead to any make target reporting the following
error as reported by Quentin in [1]:

    $ make help > /dev/null
    dirname: missing operand
    Try 'dirname --help' for more information.

Fix this by instead grep'ing for the new go:build tags in
PRIV_TEST_PKGS_EVAL.

[1] https://github.com/cilium/cilium/pull/19169#issuecomment-1096858549

Fixes: 1108684e354d ("Update Go to 1.18")

Reported-by: Quentin Monnet <quentin@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
